Responsive Curve2

XinYenFon opened the pull request:
This PR gives SMF 2.1 to handle some resolutions to display properly (aka Responsive). Admin side currently untouched and probably stay that way for now.
##### Tested with:
- Nokia Lumia 520
- Samsung Galaxy S3
- Samsung Galaxy Note2
- Samsung Galaxy S4
- Nexus7

Fixes #272
Fixes #347
Signed-off-by: Antes antes@simplemachines.org

XinYenFon committed:
Responsive Curve2
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
We don't have to do things twice.
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Some changes may change in future again...
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Little updates
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Changes;
- LinkTree now separated the unread & links
- Merged some parts with 480px & 720px

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Small update
Signed-off-by: Antes <antes@simplemachines.org>

MissAllSunday commented:
@Antes Do you wish to merge this now or do you want to add something else before merging it?

XinYenFon commented:
Actually i opened this PR for tracking and live testing with people. It currently has dependencies;

https://github.com/SimpleMachines/SMF2.1/pull/1551
https://github.com/SimpleMachines/SMF2.1/pull/1477
https://github.com/SimpleMachines/SMF2.1/pull/1471

XinYenFon committed:
Merge branch 'release-2.1' of git://github.com/SimpleMachines/SMF2.1 into responsive_paws

# By Suki (95) and others
# Via Jessica González (22) and others
* 'release-2.1' of git://github.com/SimpleMachines/SMF2.1: (143 commits)
  Wrong help popup in Admin / Search / Search Weightings fixes #1602
  ! Custom message icons will fail in PostgreSQL (fixes #1605)
  ! Typo (fixes #1601)
  Unknown column 'pmr.id_recipient' fixes #1585
  Whitespace fix
  Report post to mod permission mismatch fixes #1593
  Its not used my mistake :/ Signed-off-by: Antes <antes@simplemachines.org>
  errors needs to be in errors, while you load the errors... Signed-off-by: Antes <antes@simplemachines.org>
  $txt['registration_form'] = 'Registration Form'; needs to be in Login.lng file as well Signed-off-by: Antes <antes@simplemachines.org>
  ! Typo
  ! db_query doesn't like errors relating to trying to drop non-existent columns...
  ! The "db_alter_table" function only exists for SQLite/SQLite3 (fixes #1533)
  Fixes #1555
  Fiixes #1559
  Fixes #1572
  fixes #1564
  ! Fix several issues with allowdTo() (Fixes #1561)
  Use call_hook_helper on integrate_buffer hook
  use call_hook_helper on the smf_main()
  fixes #1548 #1549
  ...

XinYenFon committed:
some stuff to revert...
Signed-off-by: Antes <antes@simplemachines.org>

MissAllSunday commented:
I'm sorry, don't fully understand what you mean. I thought this PR had the bulk of the responsive theme and will need to be merged first before merging any other PRs.

XinYenFon commented:
Yes and no, its full work for responsive theme. No, once we change the parts of the theme, we need to update the CSS file as well. We are doing the responsive side with little less template addition and leaving all work to index.responsive.css.

For example; one of the PR changes table based styling to div based for board-index so we need to change CSS file to newly introduced systematic/classes to keep it working.

MissAllSunday commented:
I'm sorry, it is still unclear for me if this branch needs to be merged or not or if the other ones needs to be merged and if so in what order to a void unnecessary merge conflicts.

XinYenFon commented:
#1471 & #1551 already merged in, if it has no merge conflict first #1477 needs to be pull in then I'll change some stuff in this PR and it'll be ready for merge.

XinYenFon committed:
Merge branch 'release-2.1' of git://github.com/simplemachines/SMF2.1 into responsive_paws

# By John Rayes (42) and others
# Via Michael Eshom (20) and others
* 'release-2.1' of git://github.com/simplemachines/SMF2.1: (79 commits)
  i<3 my paws Signed-off-by: Antes <antes@simplemachines.org>
  ! Update language strings to reflect new functionality (fixes #1622)
  ! Remove old code (fixes #1621)
  small language mistake Signed-off-by: Antes <antes@simplemachines.org>
  Now this is the last point! Signed-off-by: Antes <antes@simplemachines.org>
  a little mistake. Signed-off-by: Antes <antes@simplemachines.org>
  Conflict is the ... problem Signed-off-by: Antes <antes@simplemachines.org>
  fix the conflict :) Signed-off-by: Antes <antes@simplemachines.org>
  The first parameter in mergePosts() was optional, while the others weren't
  Revert "This query makes no difference"
  nobbc tags wouldn't work if they encompassed multiple lines. 10 years...
  There's no 'mailq' action
  oops
  This query makes no difference
  Whitespace fix
  Improper array handling in createPost fixes #1603
  Whitespace fix
  ! unsticky a topic is recorded as sticky in moderation log
  Undefined variable: id_label fixes #1586
  Continue converting collapsed categories
  ...

XinYenFon committed:
! some little update after a long time (latest changes applied)
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Adjust me if you can!
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
buttons to hide and show... jQToggle for now
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
CSS Update

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
who?responsive
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
* Profile improvements
* Added infolink class to get rid of <br> element on profile
> Known Issue: Long mail address cause scrolling.

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
! not PX use %
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Some of the last changes before the first merge...
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
! Better search screen on small size phones
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
* Some changes on profile (edit profile) --Special thanks to @snrj

Signed-off-by: Antes <antes@simplemachines.org>

tekkla commented on Themes/default/index.template.php:
```diff
@@ -99,7 +99,8 @@ function template_html_above()
 	<meta name="keywords" content="' . $context['meta_keywords'] . '">' : '', '
-	<title>', $context['page_title_html_safe'], '</title>';
+	<title>', $context['page_title_html_safe'], '</title>
+	<meta name="viewport" content="width=device-width, initial-scale=1" />';
```
Only ">"

XinYenFon replied:
Yea html5 I'll fix it when I had time thanks rookie mistake :P

-----Original Message-----
From: "tekkla" notifications@github.com
Sent: ‎22.‎5.‎2014 22:32
To: "SimpleMachines/SMF2.1" SMF2.1@noreply.github.com
Cc: "Mert Alınbay" malinbay@gmail.com
Subject: Re: [SMF2.1] Responsive Curve2 (#1468)

In Themes/default/index.template.php:

> @@ -99,7 +99,8 @@ function template_html_above()
>   <meta http-equiv="Content-Type" content="text/html; charset=', $context['character_set'], '">
>   <meta name="description" content="', !empty($context['meta_description']) ? $context['meta_description'] : $context['page_title_html_safe'], '">', !empty($context['meta_keywords']) ? '
>   <meta name="keywords" content="' . $context['meta_keywords'] . '">' : '', '
> - <title>', $context['page_title_html_safe'], '</title>';
> - <title>', $context['page_title_html_safe'], '</title>
> - <meta name="viewport" content="width=device-width, initial-scale=1" />';
>   Only ">"
>   —
>   Reply to this email directly or view it on GitHub.

tekkla replied:
You gave me the link ;)

tekkla commented on Themes/default/index.template.php:
```diff
@@ -407,6 +408,11 @@ function template_menu()
+					<script>
+						$( "a.button.mobile" ).click(function() {
+							$( "#menu_nav" ).toggle();
+						});</script>
```
Maybe you want to shift inline scripts to a js file. It is quite easier to maintain.

XinYenFon replied:
jQ toggle only for some tests I don't see we keep it for final. We have some different ideas but maybe too soon to talk about it :)

XinYenFon committed:
Update the repo (take bug-fixes).

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Some changes;
! Stats center fix (half_content)
! Registration page improvement (Compatibility with custom fields)

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Bye superfish :)
-- and some button based fixes :P
Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Surely this little fella wants its own commit, that's the only explanation i can give to you :P

Signed-off-by: Antes <antes@simplemachines.org>

XinYenFon committed:
Conflict fix:
- Themes/default/css/index.css

Signed-off-by: Antes <antes@simplemachines.org>
